octodns_ultra/__init__.py

```diff
@@ -350,6 +350,21 @@ def populate(self, zone, target=False, lenient=False):
         )
         return exists
 
+    def _force_root_ns_update(self, changes):
+        '''
+        Changes any 'Create' changetype for a root NS record to an 'Update'
+        changetype. Used on new zone creation, since Ultra will automatically create root NS records.
+        This means our desired NS records must be applied as an Update, rather than a Create.
+        '''
+        for change in changes:
+            if (
+                change.record.name == ""
+                and isinstance(change.record, NsRecord)
+                and isinstance(change, Create)
+            ):
+                change.__class__ = Update
```

Comment:
This doesn't populate `existing`, but seems to be a bit annoying to properly do so in this case. Not sure how critical that is (doesn't look like it's used in this provider)

Reply:
Sorry I missed this comment when I looked things over last night. I'd probably lean towards instantiating a new Update change rather than altering the class of the existing one. Python would be OK with altering the class and so long as `.existing` was never used no one would be the wiser, but if something in the future tried to access it for some reason it'd blow up with a confusing error saying the attribute doesn't exist on a class who's constructor would have set it leaving people to scratch their heads if they didn't know this swap happened. 

```python
for i, change in enumerate(changes):
    ...
    # None seems to make logical sense here, though change.record would be fine for 
    # existing as well. both are a little funky
    changes[i] = Update(None, change.record)
```

Reply:
Ack! Makes sense. Thanks for the quick reviews!